test/unit/NodeCreateTransaction.js

```diff
@@ -0,0 +1,252 @@
+        const NEW_ADMIN_KEY = PrivateKey.fromStringED25519(
+            "302e020100300506032b65700422042062c4b69e9f45a554e5424fb5a6fe5e6ac1f19ead31dc7718c2d980fd1f998d4b",
+        ).publicKey;
+        tx.freeze();
+        let err = false;
+        try {
+            tx.setAdminKey(NEW_ADMIN_KEY);
+        } catch (error) {
+            err = error.toString().includes("transaction is immutable");
+        }
+
+        if (!err) {
+            throw new Error(
+                "transaction successfully built from invalid bytes",
+            );
+        }
+    });
+    it("should change certificate", function () {
+        const NEW_CERTIFICATE = Buffer.from("newCertificate");
+        tx.setGossipCaCertificate(NEW_CERTIFICATE);
+        expect(tx.gossipCaCertificate.equals(NEW_CERTIFICATE)).to.be.true;
+    });
+
+    it("should not change certificate if frozen", function () {
+        const NEW_CERTIFICATE = Buffer.from("newCertificate");
+        tx.freeze();
+        let err = false;
+        try {
+            tx.setGossipCaCertificate(NEW_CERTIFICATE);
+        } catch (error) {
+            err = error.toString().includes("transaction is immutable");
+        }
+
+        if (!err) {
+            throw new Error(
+                "transaction successfully built from invalid bytes",
+            );
+        }
+    });
+
+    it("should change certificate hash", function () {
+        const NEW_CERTIFICATE = Buffer.from("certificate_hash");
+        tx.setCertificateHash(NEW_CERTIFICATE);
+        expect(tx.certificateHash.equals(NEW_CERTIFICATE)).to.be.true;
+    });
+
+    it("should not change certificate hash if frozen", function () {
+        const NEW_CERTIFICATE = Buffer.from("certificate_hash");
+        tx.freeze();
+        let err = false;
+        try {
+            tx.setCertificateHash(NEW_CERTIFICATE);
+        } catch (error) {
+            err = error.toString().includes("transaction is immutable");
+        }
+
+        if (!err) {
+            throw new Error(
+                "transaction successfully built from invalid bytes",
+            );
+        }
+    });
+
+    it("should change gossip endpoints", function () {
+        const NEW_GOSSIP_ENDPOINTS = [
+            new ServiceEndpoint().setIpAddressV4(Uint8Array.of(127, 0, 0, 2)),
+        ];
+        tx.setGossipEndpoints(NEW_GOSSIP_ENDPOINTS);
+        expect(tx.gossipEndpoints.join("")).to.equal(
+            NEW_GOSSIP_ENDPOINTS.join(""),
+        );
+    });
+
+    it("should not change gossip endpoints if frozen", function () {
+        const NEW_GOSSIP_ENDPOINTS = [
+            new ServiceEndpoint().setIpAddressV4(Uint8Array.of(127, 0, 0, 2)),
+        ];
+        tx.freeze();
+        let err = false;
+        try {
+            tx.setGossipEndpoints(NEW_GOSSIP_ENDPOINTS);
+        } catch (error) {
+            err = error.toString().includes("transaction is immutable");
+        }
+
+        if (!err) {
```

Comment:
This is weird. Can you just do this inside the catch block?
```js
if (!error.toString().includes("transaction is immutable")) {
    throw new Error("transaction successfully built from invalid bytes");
}
```

Reply:
If it doesn't go in the catch block the test will still pass which is why we use it in all tests. It's a weird syntax, I checked if there is some mocha plugin for it that can handle these scenarios.
1. we are going to need to change tens of files with the new pattern 
2. another package to worry about 

So this is probably why the previous maintainers of the SDK used this pattern.